src/FffGcodeWriter.cpp

```diff
@@ -2379,159 +2380,187 @@ bool FffGcodeWriter::addSupportToGCode(const SliceDataStorage& storage, LayerPla
+    // Process the support islands
+    for (const auto& path: island_order_optimizer.paths)
     {
-        support_infill_angle = storage.support.support_infill_angles.at(gcode_layer.getLayerNr() % storage.support.support_infill_angles.size());
-    }
-    constexpr size_t infill_multiplier = 1; // there is no frontend setting for this (yet)
-    constexpr size_t wall_line_count = 0;
-    coord_t default_support_line_width = infill_extruder.settings.get<coord_t>("support_line_width");
-    if (gcode_layer.getLayerNr() == 0 && mesh_group_settings.get<EPlatformAdhesion>("adhesion_type") != EPlatformAdhesion::RAFT)
-    {
-        default_support_line_width *= infill_extruder.settings.get<Ratio>("initial_layer_line_width_factor");
+        const SupportInfillPart& part = *path.vertices;
+        added_something |= processSupportWalls(storage, gcode_layer, part, config);
+        added_something |= processSupportInfill(storage, gcode_layer, part, config);
     }
+    return added_something;
+}
 
-    EFillMethod support_pattern = infill_extruder.settings.get<EFillMethod>("support_pattern");
-    if (gcode_layer.getLayerNr() <= 0 && (support_pattern == EFillMethod::LINES || support_pattern == EFillMethod::ZIG_ZAG))
-    {
-        support_pattern = EFillMethod::GRID;
-    }
-    const bool zig_zaggify_infill = infill_extruder.settings.get<bool>("zig_zaggify_support");
-    constexpr bool connect_polygons = false; // polygons are too distant to connect for sparse support
-    const bool skip_some_zags = infill_extruder.settings.get<bool>("support_skip_some_zags");
-    const size_t zag_skip_count = infill_extruder.settings.get<size_t>("support_zag_skip_count");
+bool FffGcodeWriter::processSupportWalls(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SupportInfillPart& part, const SupportConfig& config) const
+{
+    bool added_something = false;
 
-    // create a list of outlines and use PathOrderOptimizer to optimize the travel move
-    PathOrderOptimizer<const SupportInfillPart*> island_order_optimizer(gcode_layer.getLastPlannedPositionOrStartingPosition());
-    for(const SupportInfillPart& part : support_layer.support_infill_parts)
+    std::vector<std::vector<std::vector<ExtrusionJunction>>> wall_toolpaths(part.wall_toolpaths.size());
+    for (const auto& toolpath : part.wall_toolpaths)
     {
-        island_order_optimizer.addPolygon(&part);
+        for (const auto& line : toolpath)
+        {
+            wall_toolpaths[line.inset_idx].emplace_back(line.junctions.begin(), line.junctions.end());
+        }
     }
-    island_order_optimizer.optimize();
 
-    //Print the thicker infill lines first. (double or more layer thickness, infill combined with previous layers)
-    for(const PathOrderOptimizer<const SupportInfillPart*>::Path& path : island_order_optimizer.paths)
+    for(const auto& path : wall_toolpaths)
     {
-        const SupportInfillPart& part = *path.vertices;
+        if(path.empty())
+            continue;
```

Comment:
Code style: Brackets around indentation.
```suggestion
        {
            continue;
        }
```

Reply:
I thought a got them all, this one slipped through. Will change it.